src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -143,11 +144,26 @@ public ImmutableMap<RepoRecordedInput.DirTree, String> getRecordedDirTreeInputs(
   @StarlarkMethod(
       name = "name",
       structField = true,
-      doc = "The name of the external repository created by this rule.")
+      doc =
+          "The canonical name of the external repository created by this rule. This name is guaranteed to be unique among all external repositories, but its exact format is not specified.")
   public String getName() {
     return rule.getName();
   }
 
+  @StarlarkMethod(
+      name = "original_name",
+      structField = true,
+      doc =
+          "The name that was originally specified as the 'name' attribute when this repository rule was instantiated. This name is not necessarily unique among external repositories (see 'name').")
+  public String getOriginalName() {
+    String originalName = (String) rule.getAttr("$original_name", Type.STRING);
+    // The original name isn't set for WORKSPACE-defined repositories as well as repositories
```

Comment:
Are these still a thing at HEAD? What's our strategy for backporting this, should I add new tests to the backports?